pkg/gardenlet/controller/seed/seed/nginxingress.go

```diff
@@ -58,11 +61,16 @@ func (r *Reconciler) newIngressDNSRecord(ctx context.Context, log logr.Logger, s
 
 func getDNSProviderSecretData(ctx context.Context, gardenClient client.Client, seed *gardencorev1beta1.Seed) (map[string][]byte, error) {
 	if dnsConfig := seed.Spec.DNS; dnsConfig.Provider != nil {
-		secret, err := kubernetesutils.GetSecretByReference(ctx, gardenClient, &dnsConfig.Provider.SecretRef)
+		credentials, err := kubernetesutils.GetCredentialsByObjectReference(ctx, gardenClient, dnsConfig.Provider.CredentialsRef)
 		if err != nil {
 			return nil, err
 		}
-		return secret.Data, nil
+		switch creds := credentials.(type) {
+		case *corev1.Secret:
+			return creds.Data, nil
+		case *securityv1alpha1.WorkloadIdentity:
+			return nil, fmt.Errorf("not supported yet") // TODO(vpnachev): Add support for WorkloadIdentity
```

Comment:
Nit: The error message can be a bit more explicit on what had gone wrong
```suggestion
			return nil, fmt.Errorf("workload identity is not supported as DNS provider credentials") // TODO(vpnachev): Add support for WorkloadIdentity
```

Reply:
Done with 126b489bb9852371b23e3340a37baeaf12738fde